src/Backend/Modules/Pages/Domain/Page/PageRepository.php

```diff
@@ -222,6 +222,71 @@ public function getFirstChild(int $pageId, string $status, string $language): ?P
             ->getOneOrNullResult();
     }
 
+    public function findOneByParentsAndUrlAndStatusAndLanguage(
+        array $parentIds,
+        string $url,
+        string $status,
+        string $language
```

Comment:
I would use the Locale value object here

Reply:
See my previous comment